<launch>
  <arg name="world_name" default="green_field"/>
  <arg name="paused" default="false"/>
  <arg name="gui" default="true"/>
  <arg name="use_sim_time" default="false"/>
  
  <include file="$(find et_gazebo)/launch/world.launch">
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)" />
  </include>

  <group ns="ET3116A">
    <include file="$(find et_gazebo)/launch/spawn_robot.launch">
      <arg name="model" value="ET3116A" />
      <arg name="namespace" value="ET3116A" />
      <arg name="is_kinematic" value="true"/>
      <arg name="x" value="-10"/>
      <arg name="y" value="0"/>
      <arg name="z" value="0"/>
      <arg name="R" value="0"/>
      <arg name="P" value="0"/>
      <arg name="Y" value="0"/>
    </include>
    <node name="et3116a_driver" pkg="et_simulator" type="et3116a_driver_node" output="screen">
    </node>
    <node name="et3116a_joy" pkg="et_simulator" type="et3116a_joy_node">
    </node>
    <node name="et3116a_servo" pkg="et_simulator" type="et3116a_servo_node" output="screen">
      <remap from="cmd_vel" to="/void"/>
    </node>
  </group>

  <group ns="dji_m100_0">
    <include file="$(find uav_gazebo)/launch/spawn_mav.launch">
      <arg name="model" value="dji_m100_default" />
      <arg name="namespace" value="dji_m100_0" />
      <arg name="is_kinematic" value="true"/>
      <arg name="x" value="0"/>
      <arg name="y" value="0"/>
      <arg name="z" value="0"/>
      <arg name="R" value="0"/>
      <arg name="P" value="0"/>
      <arg name="Y" value="0"/>
    </include>
    <node name="hil_interface" pkg="uav_simulator" type="hil_interface_node">
      <param name="model_name" type="string" value="dji_m100_0"/>
    </node>
    <node name="moving_drone" pkg="et_simulator" type="moving_drone_node">
      <param name="x_offset" type="double" value="8"/>
      <param name="omega" type="double" value="0.3"/>
      <param name="a" type="double" value="6"/>
    </node>
  </group>

  <group ns="dji_m100_1">
    <include file="$(find uav_gazebo)/launch/spawn_mav.launch">
      <arg name="model" value="dji_m100_default" />
      <arg name="namespace" value="dji_m100_1" />
      <arg name="is_kinematic" value="true"/>
      <arg name="x" value="0"/>
      <arg name="y" value="0"/>
      <arg name="z" value="0"/>
      <arg name="R" value="0"/>
      <arg name="P" value="0"/>
      <arg name="Y" value="0"/>
    </include>
    <node name="hil_interface" pkg="uav_simulator" type="hil_interface_node">
      <param name="model_name" type="string" value="dji_m100_1"/>
    </node>
    <node name="moving_drone" pkg="et_simulator" type="moving_drone_node">
      <param name="x_offset" type="double" value="2"/>
      <param name="omega" type="double" value="-0.6"/>
      <param name="a" type="double" value="3"/>
      <param name="phase" type="double" value="-2.3"/>
    </node>
  </group>

  <node name="joy" pkg="joy" type="joy_node" output="screen"/>

  <!-- <node name="image_view" pkg="image_view" type="image_view" args="image:=/camera/image_raw">
    <param name="autosize" type="bool" value="true"/>
  </node> -->

  <node name="image_view" pkg="image_view" type="image_view" args="image:=/ET3116A/image_output">
    <param name="autosize" type="bool" value="true"/>
  </node>

</launch>
